hooks/src/index.js

```diff
@@ -240,7 +239,7 @@ export function useCallback(callback, args) {
  * @param {import('./internal').PreactContext} context
  */
 export function useContext(context) {
-	const provider = currentComponent.context[context._id];
+	const provider = currentInternal._component.context[context._id];
```

Comment:
A future PR could attach `.context` on an internal (and also call provider.sub with an internal)